packages/vscode/src/plugins/prisma-language-server/index.ts

```diff
@@ -202,56 +204,15 @@ const plugin: PrismaVSCodePlugin = {
         },
       } as any, // eslint-disable-line @typescript-eslint/no-explicit-any
     }
-    // const config = workspace.getConfiguration('prisma')
-
-    workspace.onDidChangeConfiguration(
-      (event) => {
-        const fileWatcherConfigChanged = event.affectsConfiguration('prisma.fileWatcher')
-
-        if (fileWatcherConfigChanged) {
-          const isFileWatcherEnabled = workspace.getConfiguration('prisma').get('fileWatcher')
-          const rootPath = workspace.workspaceFolders?.[0].uri.path
-          // This setting defaults to true (see package.json of vscode extension)
-          if (isFileWatcherEnabled) {
-            // if watcherInstance.closed === true, the watcherInstance was closed previously and can be safely restarted
-            // if watcherInstance.closed === false, it is already running
-            // but if the JSON settings are empty like {} and the user enables the file watcherInstance
-            // we need to catch that case to avoid starting another extra file watcherInstance
-            // eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/no-unsafe-member-access
-
-            if (watcherInstance && watcherInstance.isClosed() === false) {
-              console.debug(
-                "onDidChangeConfiguration: watcherInstance.isClosed() === false so it's already running. Do nothing.",
-              )
-            } else {
-              // Let's start it
-              if (rootPath) {
-                watcherInstance = startFileWatcher(rootPath)
-                // If the file was just created
-                watcherInstance.on('add', onFileChange)
-                // If the file was modified
-                watcherInstance.on('change', onFileChange)
-                console.debug('onDidChangeConfiguration: File Watcher is now enabled and started.')
-              } else {
-                console.debug('onDidChangeConfiguration: rootPath is falsy')
-              }
-            }
-          } else {
-            // Let's stop it
-            if (watcherInstance) {
-              watcherInstance.close()
-              console.debug('onDidChangeConfiguration: File Watcher stopped.')
-            } else {
-              console.debug('onDidChangeConfiguration: No File Watcher found')
-            }
-          }
-        }
-      },
-      null,
-      context.subscriptions,
-    )
 
     context.subscriptions.push(
+      // when the file watcher settings change, we need to ensure they are applied
+      workspace.onDidChangeConfiguration((event) => {
+        if (event.affectsConfiguration('prisma.fileWatcher')) {
+          setGenerateWatcher(!!workspace.getConfiguration('prisma').get('fileWatcher'))
+        }
+      }),
```

Comment:
Why now here in `context.subscriptions.push` vs outside where this was handled before?

Reply:
It's two ways of writing the same thing.